daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -669,9 +670,70 @@ func (devices *DeviceSet) loadMetadata(hash string) *DevInfo {
 	return info
 }
 
+func getDeviceUUID(device string) (string, error) {
+	out, err := exec.Command("blkid", "-s", "UUID", "-o", "value", device).Output()
```

Comment:
I am not aware of any other easy way to do it. blkid comes from util-linux package and mkfs comes from util-linux too. As we are already using mkfs, I think it is safe to assume that blkid is available too on machines running docker.